tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ManagedExecutorTest.java

```diff
@@ -121,6 +122,48 @@ public void builderForManagedExecutorIsProvided() {
                 "MicroProfile Concurrency implementation does not provide a ManagedExecutor builder.");
     }
 
+    /**
+     * Verify that the MicroProfile Concurrency ManagedExecutor implementation clears context
+     * types that are not configured under propagated, or cleared.
+     * @throws TimeoutException 
+     * @throws ExecutionException 
+     * @throws InterruptedException 
+     */
+    @Test
+    public void clearUnspecifiedContexts() throws InterruptedException, ExecutionException, TimeoutException {
+        ManagedExecutor executor = ManagedExecutor.builder()
+                .propagated(Buffer.CONTEXT_NAME)
+                .build();
+        
+        int originalPriority = Thread.currentThread().getPriority();     
+        try {
+            // Set non-default values
+            int newPriority = originalPriority == 3 ? 2 : 3;
+            Thread.currentThread().setPriority(newPriority);
+            Buffer.set(new StringBuffer("clearUnspecifiedContexts-test-buffer-A"));
+
+            Future<Integer> future = executor.submit(() -> {
+                    Assert.assertEquals(Buffer.get().toString(), "clearUnspecifiedContexts-test-buffer-A",
+                            "Context type was not propagated to contextual action.");
+
+                    Buffer.set(new StringBuffer("clearUnspecifiedContexts-test-buffer-B"));
+
+                    return Thread.currentThread().getPriority();
+            });
+
+            Assert.assertEquals(future.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), Integer.valueOf(Thread.NORM_PRIORITY),
+                    "Context type that remained unspecified was not cleared by default.");
```

Comment:
In order to truly test that context type is cleared, use executor.completedFuture(x).thenRun(task), such that it will need to clear the context that is already established on the main test thread rather than getting a pooled thread which starts out cleared where it would be a no-op.